src/init.cpp

```diff
@@ -1184,10 +1213,12 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         } else {
             // Addrman can be in an inconsistent state after failure, reset it
             node.addrman = std::make_unique<CAddrMan>(/* deterministic */ false, /* consistency_check_ratio */ check_addrman);
+            node.addrman->m_asmap = asmap;
```

Comment:
Why making this assignment twice? The line `node.addrman->m_asmap = asmap;` is literally repeated? I think it could be done afterwards?

Reply:
Ah, I realized that's the whole point of this commit.